controllers/awsmachine_controller_unit_test.go

```diff
@@ -17,25 +17,1507 @@ limitations under the License.
+
+					instance.State = infrav1.InstanceStateStopping
+					_, _ = reconciler.reconcileNormal(context.Background(), ms, cs, cs, cs)
+					g.Expect(ms.AWSMachine.Status.InstanceState).To(PointTo(Equal(infrav1.InstanceStateStopping)))
+					g.Expect(ms.AWSMachine.Status.Ready).To(Equal(false))
+					g.Expect(buf.String()).To(ContainSubstring(("EC2 instance state changed")))
+					expectConditions(g, ms.AWSMachine, []conditionAssertion{{infrav1.InstanceReadyCondition, corev1.ConditionFalse, clusterv1.ConditionSeverityError, infrav1.InstanceStoppedReason}})
+				})
+
+				t.Run("should then set instance to stopped and unready", func(t *testing.T) {
+					g := NewWithT(t)
+					awsMachine := getAWSMachine()
+					setup(t, g, awsMachine)
+					defer teardown(t, g)
+					instanceCreate(t, g)
+					getCoreSecurityGroups(t, g)
+
+					instance.State = infrav1.InstanceStateStopped
+					_, _ = reconciler.reconcileNormal(context.Background(), ms, cs, cs, cs)
+					g.Expect(ms.AWSMachine.Status.InstanceState).To(PointTo(Equal(infrav1.InstanceStateStopped)))
+					g.Expect(ms.AWSMachine.Status.Ready).To(Equal(false))
+					g.Expect(buf.String()).To(ContainSubstring(("EC2 instance state changed")))
+					expectConditions(g, ms.AWSMachine, []conditionAssertion{{infrav1.InstanceReadyCondition, corev1.ConditionFalse, clusterv1.ConditionSeverityError, infrav1.InstanceStoppedReason}})
+				})
+
+				t.Run("should then set instance to running and ready once it is restarted", func(t *testing.T) {
+					g := NewWithT(t)
+					awsMachine := getAWSMachine()
+					setup(t, g, awsMachine)
+					defer teardown(t, g)
+					instanceCreate(t, g)
+					getCoreSecurityGroups(t, g)
+
+					instance.State = infrav1.InstanceStateRunning
+					_, _ = reconciler.reconcileNormal(context.Background(), ms, cs, cs, cs)
+					g.Expect(ms.AWSMachine.Status.InstanceState).To(PointTo(Equal(infrav1.InstanceStateRunning)))
+					g.Expect(ms.AWSMachine.Status.Ready).To(Equal(true))
+					g.Expect(buf.String()).To(ContainSubstring(("EC2 instance state changed")))
+				})
+			})
+			t.Run("deleting the AWSMachine outside of Kubernetes", func(t *testing.T) {
+				var buf *bytes.Buffer
+				deleteMachine := func(t *testing.T, g *WithT) {
+					t.Helper()
+
+					buf = new(bytes.Buffer)
+					klog.SetOutput(buf)
+					secretSvc.EXPECT().Delete(gomock.Any()).Return(nil).Times(1)
+					secretSvc.EXPECT().Create(gomock.Any(), gomock.Any()).Return("test", int32(1), nil).Times(1)
+					secretSvc.EXPECT().UserData(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil).Times(1)
+				}
+
+				t.Run("should warn if an instance is shutting-down", func(t *testing.T) {
+					g := NewWithT(t)
+					awsMachine := getAWSMachine()
+					setup(t, g, awsMachine)
+					defer teardown(t, g)
+					instanceCreate(t, g)
+					deleteMachine(t, g)
+					instance.State = infrav1.InstanceStateShuttingDown
+					_, _ = reconciler.reconcileNormal(context.Background(), ms, cs, cs, cs)
```

Comment:
Expecting deletion calls in this test but reconcileNormal() is called. 
Which scenario is this? 
Probably this question's answer is the title above but could not follow.

Reply:
I think this is to cover the scenarios where machine gets deleted with manual intervention during the AWSMachine create reconciliation.